include/onnxruntime/core/graph/graph.h

```diff
@@ -1110,6 +1110,14 @@ class Graph {
   // Graph value_info.
   std::vector<const NodeArg*> value_info_;
 
+  // Strings which have been used as node names.
+  // New node name should not conflict with this set.
+  std::unordered_set<std::string> generated_node_names_;
```

Comment:
wondering should `generated_node_names_` and `generated_node_arg_names_` be outside graph class, as those names are basically from generator tools that doesn't have direct relation with the graph. The user can just get a generated name without doing anything with the graph. So it would be a bit strange for the graph to maintain these set. Maybe make it an util function which takes a graph object as parameter?

Reply:
I am open to store outside. The current setting has a meaning --- one graph has its own namespace. Let's wait for more comments.